core/src/main/scala/org/apache/spark/rdd/PairRDDFunctions.scala

```diff
@@ -163,6 +163,38 @@ class PairRDDFunctions[K, V](self: RDD[(K, V)])
   }
 
   /**
+   * Returns the top k (largest) elements for each key from this RDD as defined by the specified
+   * implicit Ordering[T]. This does the opposite of [[takeOrderedByKey]].
+   *
+   * @param num k, the number of top elements to return
+   * @param ord the implicit ordering for T
+   * @return an RDD that contains the top k values for each key
+   */
+  def topByKey(num: Int)(implicit ord: Ordering[V]): RDD[(K, Array[V])] =
+    takeOrderedByKey(num)(ord.reverse)
+
+  /**
+   * Returns the first k (smallest) elements for each key from this RDD as defined by the specified
+   * implicit Ordering[T] and maintains the ordering. This does the opposite of [[topByKey]].
+   *
+   * @param num k, the number of elements to return
+   * @param ord the implicit ordering for T
+   * @return an RDD that contains the smallest k values for each key
+   */
+  def takeOrderedByKey(num: Int)(implicit ord: Ordering[V]): RDD[(K, Array[V])] = {
```

Comment:
I think having `topByKey` should be sufficient. Users can reverse the order easily. @rxin